src/wp-includes/class-wp-list-util.php

```diff
@@ -166,9 +166,17 @@ public function pluck( $field, $index_key = null ) {
 			foreach ( $this->output as $key => $value ) {
 				if ( is_object( $value ) ) {
 					$newlist[ $key ] = $value->$field;
-				} else {
+				} elseif ( is_array( $value ) ) {
 					$newlist[ $key ] = $value[ $field ];
 				}
+
+				if ( ! ( is_object( $value ) || is_array( $value ) ) ) {
```

Comment:
Hmm...

Starting to wonder if an `else` is more appropriate for the `_doing_it_wrong()` condition. That saves on some checks.

i.e.
Currently:
- Good object data: `if is object` + `if not object and not array` (2)
- Good array data: `if is object` + `else if is array` + `if not object and not array` (3)
- Bad data: `if is object` + `else if is array` + `if not object and not array` (3)

With `else`:
- Good object data: `if is object` (1)
- Good array data: `if is object` + `else if is array` (2)
- Bad data: `if is object` + `else if is array` + `else` (3ish)